lib/agent0/agent0/hyperdrive/interactive/i_hyperdrive.py

```diff
@@ -0,0 +1,532 @@
+        rng_seed: int | None, optional
+            The seed for the random number generator. Defaults to None.
+        rng: Generator | None, optional
+            The experiment's stateful random number generator. Defaults to creating a generator from
+            the provided random seed if not set.
+        crash_log_level: int, optional
+            The log level to log crashes at. Defaults to critical.
+        log_to_rollbar: bool, optional
+            Whether to log crash reports to rollbar. Defaults to False.
+        rollbar_log_prefix: str | None, optional
+            The prefix to prepend to rollbar exception messages.
+        crash_report_additional_info: dict[str, Any] | None, optional
+            Additional information to include in the crash report.
+        """
+
+        preview_before_trade: bool = False
+        rng_seed: int | None = None
+        rng: Generator | None = None
+        log_to_rollbar: bool = False
+        rollbar_log_prefix: str | None = None
+        crash_log_level: int = logging.CRITICAL
+        crash_report_additional_info: dict[str, Any] | None = None
+
+        def __post_init__(self):
+            if self.rng is None:
+                self.rng = np.random.default_rng(self.rng_seed)
+
+    class Addresses(HyperdriveAddresses):
+        """The addresses class that defines various addresses for Hyperdrive."""
+
+        # Subclass from the underlying addresses dataclass
+        # We simply define a class method to initialize the address from
+        # artifacts uri
+
+        @classmethod
+        def from_artifacts_uri(cls, artifacts_uri: str) -> IHyperdrive.Addresses:
+            """Builds hyperdrive addresses from artifacts uri.
+
+            Arguments
+            ---------
+            artifacts_uri: str
+                The uri of the artifacts server from which we get addresses.
+                E.g., `http://localhost:8080`.
+
+            Returns
+            -------
+            IHyperdrive.Addresses
+                The hyperdrive addresses object
+            """
+            out = fetch_hyperdrive_address_from_uri(artifacts_uri)
+            return cls._from_ethpy_addresses(out)
+
+        @classmethod
+        def _from_ethpy_addresses(cls, addresses: HyperdriveAddresses) -> IHyperdrive.Addresses:
+            return IHyperdrive.Addresses(**asdict(addresses))
+
+    def __init__(
+        self,
+        chain: IChain,
+        hyperdrive_addresses: Addresses,
+        config: Config | None = None,
+    ):
+        if config is None:
+            self.config = self.Config()
+        else:
+            self.config = config
+
+        # Define agent0 configs with this setup
+        # TODO currently getting the path based on this file's path
+        # This requires the entire monorepo to be check out, and will likely not work when
+        # installing agent0 by itself.
+        # This should get fixed when abis are exported in hypertypes.
```

Comment:
pls add a reference to this issue: https://github.com/delvtech/agent0/issues/1125

would also be great if you cross-referenced this line of code in the issue once the PR is merged.